# EBDC 2023

In [1]:
import pandas as pd

In [2]:
#!mamba install openpyxl -y

In [3]:
data = pd.read_excel("videos-ebdc-2023.xlsx",
                     sheet_name=None)

In [4]:
all_videos = []

for sheet in data.items():
    channel_name, sheet_individual = sheet
    for idx, video_title in enumerate(sheet_individual["videoTitle"]):
        video_id = sheet_individual["videoId"][idx]
        all_videos.append(
            (f"https://www.youtube.com/watch?v={video_id}",
             channel_name,
             video_title)
        )

* Nomeando as colunas.

In [5]:
data_proc = pd.DataFrame(all_videos,
                         columns=["address",
                                  "channelTitle",
                                  "videoTitle"])

In [6]:
print(len(data_proc))

408


* Universo de 408 vídeos: amostra de 198 vídeos, 95% de confiança, margem de erro de 5%.

In [7]:
data_sample = data_proc.sample(n=198,
                               random_state=42,
                               ignore_index=True)

* Gravando dados em CSV.

In [8]:
data_sample.to_csv("videos_amostra-ebdc-2023.csv")